File: java/pr9/src/main/java/com/company/pr9/entity/Route.java
package com.company.pr9.entity;

import io.jmix.core.entity.annotation.JmixGeneratedValue;
import io.jmix.core.metamodel.annotation.InstanceName;
import io.jmix.core.metamodel.annotation.JmixEntity;

import javax.persistence.*;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.UUID;
import java.util.List;

@JmixEntity
@Table(name = "TR_ROUTE")
@Entity(name = "tr_Route")
public class Route {
    @JmixGeneratedValue
    @Column(name = "ID", nullable = false)
    @Id
    private UUID id;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    @InstanceName
    @Column(name = "DEPARTURE")
    private String departure;

    public String getDeparture() {
        return departure;
    }

    public void setDeparture(String departure) {
        this.departure = departure;
    }

    @Column(name = "DESTINATION")
    private String destination;

    public String getDestination() {
        return destination;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "STATION_ID")
    private Station station;

    public Station getStation() {
        return station;
    }

    public void setStation(Station station) {
        this.station = station;
    }

    @ManyToMany(mappedBy = "routes")
    private List<Bus> buses;

    public List<Bus> getBuses() {
        return buses;
    }

    public void setBuses(List<Bus> buses) {
        this.buses = buses;
    }

}

File: java/pr9/src/main/kotlin/com/company/pr9/entity/Route.kt
package com.company.pr9.entity

import io.jmix.core.entity.annotation.JmixGeneratedValue
import io.jmix.core.metamodel.annotation.JmixEntity
import java.util.*
import javax.persistence.*

@JmixEntity
@Table(name = "ROUTE")
@Entity
open class Route {
    @JmixGeneratedValue
    @Column(name = "ID", nullable = false)
    @Id
    private val id: UUID? = null

    @Column(name = "DEPARTURE_POINT", nullable = false)
    private val departurePoint: String? = null

    @Column(name = "ARRIVAL_POINT", nullable = false)
    private val arrivalPoint: String? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "STATION_ID")
    private val station: Station? = null

    @ManyToMany(mappedBy = "routes")
    private val buses: List<Bus> = ArrayList() // Constructors, getters, setters
}
